### PR TITLE
Add first draft of the Pull Request workflow documentation.

### PR DESCRIPTION
**Pull request by adbridge**

### CODE REVIEW

**Comment by adbridge** [comment]
@sg- @AnotherButler please review

File: docs/reference/contributing/guidelines/workflow.md [hunk 1]
```diff
@@ -49,3 +49,47 @@ Pull requests on GitHub have to meet the following requirements in order to keep
...
+
+#### GitHub pull requests workflow
+
+Each pull request will go through the following workflow:
+
+<span class="images">![](https://s3-us-west-2.amazonaws.com/mbed-os-docs-images/workflow.jpg)<span>The workflow of merging a pull request</span></span>
+
+##### Pull request states
+
+Labels that the Mbed OS maintainers add to a pull request represent the pull request workflow states. The Mbed OS maintainers are responsible for moving pull requests through the workflow states.
```
Comment (AnotherButler) [inline docs/reference/contributing/guidelines/workflow.md]: Could you please create a sentence or two explaining who the Mbed OS maintainers are and what they do at the start of this section?

File: docs/reference/contributing/guidelines/workflow.md [hunk 2]
```diff
@@ -8,7 +8,7 @@ All code changes and additions to Mbed OS are handled through GitHub. If you wan
 
 Before contributing an enhancement (new feature, new port and so on) please [discuss it on the forums](https://os.mbed.com/forum/) to avoid duplication of work, as we or others might be working on a related feature.
```
Comment (sg-) [inline docs/reference/contributing/guidelines/workflow.md]: Lets create a specific forum for this ahead of time and link directly

File: docs/reference/contributing/guidelines/workflow.md (hunk 1 above)
Comment (0xc0170) [reply docs/reference/contributing/guidelines/workflow.md]: Shall we name them, and provide github names?

File: docs/reference/contributing/guidelines/workflow.md [hunk 3]
```diff
@@ -49,3 +49,47 @@ Pull requests on GitHub have to meet the following requirements in order to keep
...
 * Because we use GitHub and explicit CLAs, special commit tags that other projects may use, such as “Reviewed-by”, or “Signed-off-by”, are redundant and should be omitted. GitHub keeps track of who reviewed what and when, and our stack of signed CLAs shows us who has agreed to our development contribution agreement.
 * Prefixing your commit message with a domain is acceptable and recommended where it makes sense to do so. However, prefixing one's domain with the name of the repo is not useful. For example, making a commit entitled "mbed-drivers: Fix doppelwidget frobulation" to the `mbed-drivers` repo would not be acceptable, as it is already understood that the commit applies to `mbed-drivers`. Renaming the commit to "doppelwidget: Fix frobulation" would be better, if we presume that "doppelwidget" is a meaningful domain for changes, as it communicates that the change applies to the doppelwidget area of mbed-drivers.
+
+#### GitHub pull requests workflow
+
+Each pull request will go through the following workflow:
+
+<span class="images">![](https://s3-us-west-2.amazonaws.com/mbed-os-docs-images/workflow.jpg)<span>The workflow of merging a pull request</span></span>
```
Comment (0xc0170) [inline docs/reference/contributing/guidelines/workflow.md]: The flow is a bit complicated, can we simplify it ?

Comment (adbridge) [reply docs/reference/contributing/guidelines/workflow.md]: I think in hindsight we don't need the CLA in there as states, I can document that separately and then make the flow more logical...

File: docs/reference/contributing/guidelines/workflow.md (hunk 1 above)
Comment (adbridge) [reply docs/reference/contributing/guidelines/workflow.md]: Seems reasonable, the only reason I didn't specifically name them is in the future the list may change/grow etc and thus mean updating this documentation. By referring to the github group instead this covers them. But I am happy to do so.

Comment (adbridge) [reply docs/reference/contributing/guidelines/workflow.md]: Your changes are fine, thanks @AnotherButler

**Comment by adbridge** [comment]
@sg- @0xc0170 @AnotherButler Review comments addressed, please re-review

**Comment by AnotherButler** [comment]
@adbridge 👍 I like the addition. Please remember to address sg-'s comment about the forums before we merge.

**Comment by adbridge** [comment]
@AnotherButler Ahh I thought he was talking to you about the forums as it wasn't something I wrote ! But I can have a look into it anyway.

File: docs/reference/contributing/guidelines/workflow.md (hunk 2 above)
Comment (adbridge) [reply docs/reference/contributing/guidelines/workflow.md]: So we already have 
https://os.mbed.com/forum/mbed/
https://os.mbed.com/forum/bugs-suggestions/

Are we saying we should also have something like
https://os.mbed.com/forum/new-features  ?

Comment (adbridge) [reply docs/reference/contributing/guidelines/workflow.md]: Or should we just link directly to one of the latter two forums ?

Comment (sg-) [reply docs/reference/contributing/guidelines/workflow.md]: I think bugs and suggestions is good.

**Comment by adbridge** [comment]
@AnotherButler @sg- Specific forum links added. This should be ready now.

**Comment by adbridge** [comment]
@sg- Final graphic? Is there are specific way the flowchart needs to be then ? I will email him.

**Comment by sg-** [comment]
I'd be great to have James F do something. If not, @c1728p9 found draw.io and we could get a template started for consistency. I sent one by email.

**Comment by adbridge** [comment]
I spoke to James F and he is loaded at the moment with web work. I can have a go at reproducing using draw.io .

**Comment by adbridge** [comment]
@AnotherButler @sg- OK I have updated the diagram based on the email discussions , some additional tweaks to the doc and have rebased. Could you two please have a final once over (Amanda there are a couple of links for you to update, firstly is add one for the CLA where i tagged your name and second update the workflow pic to the new one - workflow.png). Thanks

**Pull request merged**